packages/block-library/src/columns/edit.native.js

```diff
@@ -274,7 +274,7 @@ function ColumnsEditContainer( {
 						orientation={
 							columnsInRow > 1 ? 'horizontal' : undefined
 						}
-						horizontal={ true }
+						horizontal={ columnsInRow > 1 }
```

Comment:
`numColumns` is React Native's recommended over the unsupported `flexWrap`.  However, setting `horizontal={ true }` causes `numColumns` to be ignored. This meant it was not correctly enforcing a single column.